packages/lmi/tests/test_llms.py

```diff
@@ -716,15 +716,32 @@ async def test_openrouter_reasoning(self) -> None:
         assert results[0].reasoning_content
 
         outputs: list[str] = []
-        with pytest.raises(
-            NotImplementedError,
-            match=r"Reasoning with OpenRouter is not supported in streaming mode.*",
-        ):
-            # NOTE: We invoke the call with a callback to test the streaming
-            #  from LiteLLMModel.acompletion_iter().
-            # Because it is still not supported for OpenRouter models, we test
-            #  that it raises a NotImplementedError.
-            await llm.call(messages, callbacks=[outputs.append])
+        # NOTE: We invoke the call with a callback to test the streaming
+        #  from LiteLLMModel.acompletion_iter().
+        #     Reasoning with OpenRouter is not supported in streaming mode.
+        #     https://github.com/BerriAI/litellm/issues/8631
+        #     Consider using LiteLLMModel(name='deepseek/deepseek-reasoner') instead.
+        # Because it is still not supported for OpenRouter models, we assert
+        #  that the reasoning content is not present in the result.
+        results = await llm.call(messages, callbacks=[outputs.append])
+        assert isinstance(results, list)
+        assert isinstance(results[0], LLMResult)
+        assert outputs
+
+    @pytest.mark.vcr(match_on=[*VCR_DEFAULT_MATCH_ON, "body"])
+    @pytest.mark.asyncio
+    @pytest.mark.xfail(
+        reason="LiteLLM still doesn't support OpenRouter reasoning with streaming"
+    )
```

Comment:
Can you put an exception type to ensure this is accurately aimed? Think how we provide an exception type to `pytest.raises`